.github/workflows/macos.yml

```diff
@@ -71,9 +84,38 @@ jobs:
 
       - run: make
 
+      # Launchable CLI requires Python and Java
+      # https://www.launchableinc.com/docs/resources/cli-reference/
+      - name: Set up Python
+        uses: actions/setup-python@871daa956ca9ea99f3c3e30acb424b7960676734 # v5.0.0
+        with:
+          python-version: "3.10"
+      - name: Set up Java
+        uses: actions/setup-java@7a445ee88d4e23b52c33fdc7601e40278616c7f8 # v4.0.0
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+      - name: Set up Launchable
+        id: setup_launchable
+        run: |
+          set -ex
+          pip install launchable
+          launchable verify
+          : # The build name cannot include a slash, so we replace the string here.
+          github_ref=`echo ${{github.ref}} | sed 's/\//_/g'`
+          : # With the --name option, we need to configure a unique identifier for this build.
+          : # To avoid setting the same build name as the CI which runs on other branches, we use the branch name here.
+          launchable record build --name ${github_ref}_${GITHUB_PR_HEAD_SHA}
+          echo "TEST_WITH_LAUNCHABLE=--launchable-test-reports=${GITHUB_RUN_ID}.json" >> "$GITHUB_ENV"
+        if: ${{ github.repository == 'ruby/ruby' && (matrix.test_task == 'check' || matrix.test_task == 'test-all') }}
+
+      - name: Set repeat-count option for make ${{ matrix.test_task }}
+        run: echo "TEST_WITH_LAUNCHABLE=$TEST_WITH_LAUNCHABLE --repeat-count=2" >> "$GITHUB_ENV"
+        if: ${{ matrix.test_task == 'test-all' }}
+
       - name: make ${{ matrix.test_task }}
         run: |
-          make -s ${{ matrix.test_task }} ${TESTS:+TESTS=`echo "$TESTS" | sed 's| |$$/ -n!/|g;s|^|-n!/|;s|$|$$/|'`}
+          make -s ${{ matrix.test_task }} TESTS="$TEST_WITH_LAUNCHABLE" ${TESTS:+TESTS=`echo "$TESTS" | sed 's| |$$/ -n!/|g;s|^|-n!/|;s|$|$$/|'`}
```

Comment:
I think this code has a possibility to have `TESTS` make argument 2 times below, when the `$TESTS` is unset.

```
make -s ${{ matrix.test_task }} TESTS="..." TESTS="..."
```

It seems that the logic was added by the commit 8794cc62899c6447fa4451489b9e308e2a890595. And I don't understand the logic well. So I am asking @nobu about it.

Reply:
@ono-max perhaps, the `TESTOPTS=TESTOPTS` below may work safely with existing logic: `TESTS=...`. You can check this part: https://github.com/ruby/ruby/blob/ae3e82a5259893b4d68e2bedb675c160bcd1a173/common.mk#L1907C1-L1908C79 as a reference.

```
make -s ${{ matrix.test_task }} TESTOPTS="$TESTOPTS" ${TESTS:+...}
```

Reply:
Hmm, the error occurred when using `$TESTOPTS`. https://github.com/ruby/ruby/actions/runs/7969517300/job/21755460769#step:12:42

Reply:
Sorry, this PR's CI is failing with the `TESTOPTS="$TESTOPTS"` on the latest commits in this `make check` cases. 
https://github.com/ruby/ruby/actions/runs/7969517300/job/21755460769?pr=9777#step:12:42

The reason seems that the `make check` executes the `btest-ruby` make target internally. 

```
btest-ruby
  2024-02-20 06:51:57 +0000
  Driver is ruby 3.4.0dev (2024-02-20T06:46:22Z pull/9777/merge ee2c2b2d9e) [x86_64-darwin21]
  Target is ruby 3.4.0dev (2024-02-20T06:46:22Z pull/9777/merge ee2c2b2d9e) [x86_64-darwin21]

   ../src/bootstraptest/runner.rb:283:in 'Kernel#load': cannot load such file -- /Users/runner/work/ruby/ruby/build/--launchable-test-reports=7969517300.json (LoadError)
  	from ../src/bootstraptest/runner.rb:283:in 'block in Object#load_test'
  	from <internal:array>:52:in 'Array#each'
  	from ../src/bootstraptest/runner.rb:282:in 'Object#load_test'
  	from ../src/bootstraptest/runner.rb:334:in 'Object#exec_test'
  	from ../src/bootstraptest/runner.rb:269:in 'block in Object#main'
  	from ../src/bootstraptest/runner.rb:757:in 'block (2 levels) in Object#in_temporary_working_directory'
  	from ../src/bootstraptest/runner.rb:756:in 'Dir.chdir'
  	from ../src/bootstraptest/runner.rb:756:in 'block in Object#in_temporary_working_directory'
  	from /Users/runner/work/ruby/ruby/src/lib/tmpdir.rb:99:in 'Dir.mktmpdir'
  	from ../src/bootstraptest/runner.rb:755:in 'Object#in_temporary_working_directory'
  	from ../src/bootstraptest/runner.rb:268:in 'Object#main'
  	from ../src/bootstraptest/runner.rb:801:in '<main>'
  make: *** [yes-btest-ruby] Error 1
  Error: Process completed with exit code 2.
```

Seeing the `common.mk` btest-ruby make's target (actually `uncommoon.mk` generated from the `common.mk` is used in the proces) is below. Maybe we need to set the `$(BTESTS)` when we use the `$(TESTOPTS)`. So, I don't think using the make's argument `TESTOPTS=something` is a right approach in this case.

https://github.com/ruby/ruby/blob/2a4b6ed37cbb7acb622703be4db770457c2b72cc/common.mk#L882-L887

```
$(Q)$(gnumake_recursive)$(exec) $(RUNRUBY) "$(srcdir)/bootstraptest/runner.rb" --ruby="$(PROGRAM) -I$(srcdir)/lib $(RUN_OPTS)" $(OPTS) $(TESTOPTS) $(BTESTS)
```

I have an idea about how to implement this part. That is to refactor the part of the existing `$TESTS` logic that is a logic of `skipped_tests`, or drop the logic. I need @nobu's help for this part. Currently there are 3 GitHub Actions files implementing the `skipped_tests` logic.

```
$ grep -lr skipped_tests .github/workflows/
.github/workflows/yjit-macos.yml
.github/workflows/macos.yml
.github/workflows/ubuntu.yml
```

In the mean time, you can keep this PR's code with the `TESTS="$TESTOPTS"` or try to refactor by yourself.

Reply:
@ono-max I merged the PR #10029. Now you can rebase this PR on the latest master branch. Seeing the Motivation part in my first comment on the PR, you can imagine how to implement.

Reply:
Updated

Reply:
Thanks! The updated code looks good to me.

Reply:
I confirmed that the `TESTS:  --launchable-test-reports=launchable_reports.json --repeat-count=2` is used in the `make test-all` in this PR's macOS case.
https://github.com/ruby/ruby/actions/runs/8003735114/job/21859690713?pr=9777#step:14:44